ceph_iscsi_config/gateway.py

```diff
@@ -194,15 +193,16 @@ def enable_active_tpg(self, config):
                 else:
                     break
 
-    def clear_config(self):
+    def clear_config(self, target_disks):
         """
         Remove the target definition form LIO
         :return: None
         """
         # check that there aren't any disks or clients in the configuration
         lio_root = root.RTSRoot()
 
-        disk_count = len([disk for disk in lio_root.storage_objects])
+        disk_count = len([disk for disk in lio_root.storage_objects
+                          if disk.name in target_disks])
```

Comment:
I think here too just loop over the disks specific to the target being cleared. I think you can just merge it into the tpg loop and check for luns under the tpg since that would detect if there are disks possibly being accessed through the target still.

Reply:
Fixed